docs/guide/extending/extending_thumbnails.md

```diff
@@ -0,0 +1,127 @@
+# Extending thumbnails
+
+Thumbnails allow you to easily choose an image for a specific Content Type.
+
+## Thumbnail mechanism 
+
+Thumbnail mechanism has two layers, and each layer can have many implementations.
+The mechanism checks if any of the implementations contains an image.
+If found, the image is returned as a Content Type thumbnail.
+
+First layer of the mechanism contains strategy pattern that focuses on finding a thumbnail.
+The thumbnail can be found inside or outside the Content Type
+e.g. for users thumbnails can be downloaded from avatar generating service.
+
+For this layer there are two default implementations.
+First looks for images in Field Types, and after that the second one will look for them in Content Type.
+If there are no Field Types with a thumbnail image, the `ContentType` icon will be used as a thumbnail.
+This mechanism can be modified to fit your site needs, so you can decide from where and how the thumbnails will be downloaded.
+
+Second layer of Thumbnail mechanism enables selection of thumbnail from a Field Type. 
```

Comment:
```suggestion
Second layer of thumbnail mechanism enables selection of thumbnail from a Field. 
```
Also, does this mean the second layer is used only when the first one doesn't work? Does it mean that the checkbox can be overridden by the first layer?